### PR TITLE
Send Query Parameters in client requests

### PR DESCRIPTION
**Pull request by glademiller**

### CODE REVIEW

File: src/client/writer.rs [hunk 1]
```diff
@@ -112,7 +112,9 @@ impl HttpClientWriter {
 
             // status line
             let _ = write!(buffer, "{} {} {:?}\r\n",
-                           msg.method(), msg.uri().path(), msg.version());
+                           msg.method(), 
+                           msg.uri().path_and_query().map(|u| u.as_str()).unwrap_or(""),
```
Comment (fafhrd91) [inline src/client/writer.rs]: should we use .unwrap_or("/") instead of .unwrap_or("")  ?

Comment (glademiller) [reply src/client/writer.rs]: I was trying to match the functionality here https://docs.rs/http/0.1.5/src/http/uri/mod.rs.html#339 IE before we called path and it returned "" if there was no path.

Comment (fafhrd91) [reply src/client/writer.rs]: i think empty string is a bug

Comment (glademiller) [reply src/client/writer.rs]: I could see how it could be a problem. I'll return "/"

**Review by fafhrd91 — APPROVED** [review]

**Comment by fafhrd91** [comment]
thanks!

**Pull request merged**